IPIP/0379-delegated-naming-http-api.md

```diff
@@ -0,0 +1,83 @@
+# IPIP-379: Delegated Naming HTTP API
+
+- Start Date: 2023-02-13
+- Related Issues:
+  - https://github.com/ipfs/specs/issues/343
+  - https://github.com/ipfs/specs/pull/337
+  - https://github.com/ipfs/specs/pull/377
+
+## Summary
+
+This IPIP specifies a set of HTTP APIs to offload naming system onto aother process or server.
+
+## Motivation
+
+Expanding on the motivations of [IPIP-337 HTTP Delegated Routing](0337-delegated-routing-http-api.md), the work here concentrates on delegation of _naming system_ over HTTP APIs. Naming is part of the core IPFS DHT functionality.
+The performance of naming system over the IPFS DHT can suffer from long delays due to churn of records and quorum requirements.
+
+## HTTP API Specification
+
+See [Delegated Naming HTTP API](../routing/DELEGATED_NAMING_HTTP.md) specification.
+
+## Design rationale
+
+The rationale for delegated naming over HTTP APIs closely follows the reasoning listed in [IPIP-337](0337-delegated-routing-http-api.md#design-rationale).
+
+The document proposes the following:
+- Use of HTTP semantics for publication and resolution of naming records.
+- Preference for human-readable request and response encoding, such as JSON format
+- Optional backward compatibility support for the existing ProtocolBuffer format using `Content-Type: application/x-protobuf; messageType="IpnsEntry"`
+- Use of extra headers in `OPTIONS` response to communicate the supported cpabilities and limitations, namely:
+  - `X-Ipns-Allow-Max-Size` -- to signal maximum supported IPNS record size
+  - `X-Ipns-Allow-Protobuf` -- to signal whether the server supports ProtocolBuffer formatted records.
+- Streaming interaction is not supported.
+
+### User benefit
+
+The ability of offload naming onto another process or server via an idiomatic and simple HTTP API opens up an opportunity to not only reduce the burden of naming system management, but also introduce alternative naming systems across other routing systems, namely IPNI.
+
+### Compatibility
+
+#### Backwards Compatibility
+
+##### Serialization Format
+
+IPNS records use [ProtocolBuffer serialisation format](../ipns/IPNS.md#record-serialization-format).
+This format is widely in use in IPNS over PubSub and DHT routing systems.
+One of the motivations of this document is to introduce simple to use HTTP APIs and ultimately reduce barrier for interaction across alternative systems.
+Further, interoperability across the existing and HTTP APIs is also desirable in order to reduce the barrier for adoption of the delegated HTTP APIs.
+
+The specification here maintains backwards compatibility in terms of record serialisation, with preference for human-readable formats such as JSON.
+To maintain backwards compatibility, the implementers of the HTTP API should accept requests with following `Content-Type` header:
+- `application/x-protobuf; messageType="IpnsEntry"`
```

Comment:
I'd like to push for this please:
* I think it is fair to say that the point of specs like this is to introduce extensibility. IPNS records creation delegated via this spec will most likely be unpacked. Because the whole point is to offer alternative naming system management to the DHT. So keeping the records format untouched such that records as they are could also be put onto the existing DHT interface seems unnecessary.
* The spec proposed here offers the best of both worlds: old records are optionally accepted by the server, while the new format offers an opportunity to not only improve UX when it comes to assembling IPNS records but also improve readability and cross implementation extensibility.

Reply:
Having both sgtm, as long both have the same signature bytes.
IPNI is free to take `application/vnd.ipfs.ipns-record`, and store each field once.